optuna/visualization/_rank.py

```diff
@@ -0,0 +1,396 @@
+            figure.update_xaxes(type="category")
+        if sub_plot_info.yaxis.is_cat:
+            figure.update_yaxes(type="category")
+
+        if sub_plot_info.xaxis.is_log:
+            log_range = [math.log10(p) for p in sub_plot_info.xaxis.range]
+            figure.update_xaxes(range=log_range, type="log")
+        if sub_plot_info.yaxis.is_log:
+            log_range = [math.log10(p) for p in sub_plot_info.yaxis.range]
+            figure.update_yaxes(range=log_range, type="log")
+    else:
+        figure = make_subplots(
+            rows=len(params),
+            cols=len(params),
+            shared_xaxes=True,
+            shared_yaxes=True,
+            horizontal_spacing=0.08 / len(params),
+            vertical_spacing=0.08 / len(params),
+        )
+
+        figure.update_layout(layout)
+        for x_i, x_param in enumerate(params):
+            for y_i, y_param in enumerate(params):
+                scatter = _get_rank_subplot(
+                    sub_plot_infos[y_i][x_i], info.target_name, info.has_custom_target
+                )
+                figure.add_trace(scatter, row=y_i + 1, col=x_i + 1)
+
+                xaxis = sub_plot_infos[y_i][x_i].xaxis
+                yaxis = sub_plot_infos[y_i][x_i].yaxis
+                figure.update_xaxes(range=xaxis.range, row=y_i + 1, col=x_i + 1)
+                figure.update_yaxes(range=yaxis.range, row=y_i + 1, col=x_i + 1)
+
+                if xaxis.is_cat:
+                    figure.update_xaxes(type="category", row=y_i + 1, col=x_i + 1)
+                if yaxis.is_cat:
+                    figure.update_yaxes(type="category", row=y_i + 1, col=x_i + 1)
+
+                if xaxis.is_log:
+                    log_range = [math.log10(p) for p in xaxis.range]
+                    figure.update_xaxes(range=log_range, type="log", row=y_i + 1, col=x_i + 1)
+                if yaxis.is_log:
+                    log_range = [math.log10(p) for p in yaxis.range]
+                    figure.update_yaxes(range=log_range, type="log", row=y_i + 1, col=x_i + 1)
+
+                if x_i == 0:
+                    figure.update_yaxes(title_text=y_param, row=y_i + 1, col=x_i + 1)
+                if y_i == len(params) - 1:
+                    figure.update_xaxes(title_text=x_param, row=y_i + 1, col=x_i + 1)
+    sorted_zs = np.sort(info.zs)
+    tick_coloridxs = [0, 0.25, 0.5, 0.75, 1]
+    tick_values = np.quantile(sorted_zs, tick_coloridxs)
+    rank_text = ["min.", "25%", "50%", "75%", "max."]
+    ticktext = [f"{rank_text[i]} ({tick_values[i]:3g})" for i in range(len(tick_values))]
+
+    colormap = "RdYlBu_r"
+    colorbar_trace = go.Scatter(
+        x=[None],
+        y=[None],
+        mode="markers",
+        marker=dict(
+            line={"width": 0.5, "color": "Grey"},
+            colorscale=colormap,
+            showscale=True,
+            cmin=0,
+            cmax=1,
+            colorbar=dict(thickness=10, tickvals=tick_coloridxs, ticktext=ticktext),
```

Comment:
How about adding the title?

```suggestion
            colorbar=dict(thickness=10, tickvals=tick_coloridxs, ticktext=ticktext, title=f"Rank ({info.target_name})"),
```

Reply:
<img width="442" alt="image" src="https://user-images.githubusercontent.com/24619922/228113561-09bcc861-f289-400f-8d1a-1b60e45c6a41.png">

In my opinion, this looks a bit too redundant and noisy.

Reply:
Hmmm, but without that, isn't it hard to understand what the numbers mean?

Reply:
What about changing the title of the plot to `Rank (Objective Value)`?

Reply:
That seems reasonable!

Reply:
@not522 PTAL.